inc/event_target.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <vector>
+#include "event_listener.hpp"
+
+namespace dragonslave {
+
+
+template <typename EventType>
+class EventTarget
+{
+public:
+    void add_event_listener(EventListener<EventType> listener);
+    void remove_event_listener(EventListener<EventType> listener);
+    void emit_event();
```

Comment:
emit event should take in an object of const EventType& (to have all it's listener's handle)